trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/rest/TestProxiedErrorResponses.java

```diff
@@ -95,12 +99,21 @@ public Builder filter(Builder builder)
     }
 
     @Inject
-    public TestProxiedErrorResponses(S3Client internalClient, TestingRemoteS3Facade delegatingFacade, @ForErrorResponseTest TestingHttpServer httpErrorResponseServer)
+    public TestProxiedErrorResponses(TestingRemoteS3Facade delegatingFacade, @ForErrorResponseTest TestingHttpServer httpErrorResponseServer, @ForTesting Credentials testingCredentials)
     {
-        this.internalClient = requireNonNull(internalClient, "internal client is null");
+        internalClient = clientBuilder(httpErrorResponseServer.getBaseUrl())
+                .forcePathStyle(true)
+                .credentialsProvider(() -> AwsSessionCredentials.create(testingCredentials.emulated().accessKey(), testingCredentials.emulated().secretKey(), testingCredentials.emulated().session().orElse("")))
+                .build();
         delegatingFacade.setDelegate(new PathStyleRemoteS3Facade((_, _) -> httpErrorResponseServer.getBaseUrl().getHost(), false, Optional.of(httpErrorResponseServer.getBaseUrl().getPort())));
     }
 
+    @AfterEach
```

Comment:
Is the constructor invoked once per test? It's not obvious from the test whether it's @TestInstance(PER_CLASS) or @TestInstance(PER_METHOD). I'd suggest annotating the class to make it explicit.

Reply:
`@TrinoAwsProxyTest` implies per class (and other settings).

Reply:
Hmm... if its `@TestInstance(PER_CLASS)`, then it's not going to work with this shutdown method, which is going to be called after each individual test. You need to use `@AfterAll`, instead.